### PR TITLE
[DEV-4987] refactor gql teamtree model

### PR DESCRIPTION
**Pull request by gaetano-guerriero**

### CODE REVIEW

File: server/athenian/api/internal/team_tree.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    RootTeamNotFoundError,
+    TeamNotFoundError,
+)
+from athenian.api.models.state.models import Team
+from athenian.api.models.web import TeamTree
+from athenian.api.tracing import sentry_span
+
+
+@sentry_span
+def build_team_tree_from_rows(rows: Sequence[Row], root_team_id: Optional[int]) -> TeamTree:
```
Comment (gaetano-guerriero) [inline server/athenian/api/internal/team_tree.py:16]: just moved outside of align

File: server/athenian/api/models/web/base_model_.py [hunk 2]
```diff
@@ -26,6 +33,11 @@ def __new__(mcs, name, bases, dikt, sealed=True):
             attribute_map.update(dikt["attribute_map"])
```
Comment (vmarkovtsev) [inline server/athenian/api/models/web/base_model_.py:33]: missing: `default_values.update(dikt["default_values"])`

File: server/athenian/api/models/web/team_tree.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    attribute_types = {
+        "id": int,
+        "name": str,
+        "members_count": int,
+        "total_teams_count": int,
+        "total_members_count": int,
+        "children": list[Model],  # list[TeamTree],
+        "members": list[int],
+        "total_members": list[int],
+    }
```
Comment (vmarkovtsev) [inline server/athenian/api/models/web/team_tree.py:19]:
```suggestion
    id: int
    name: str
    ...
```

File: server/athenian/api/models/web/team_tree.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        """Get the number of teams included in the team tree."""
+        return self._total_teams_count
+
+    @property
+    def total_members_count(self) -> int:
+        """Get the number of team members included in the team tree."""
+        return self._total_members_count
+
+    @property
+    def children(self) -> list["TeamTree"]:
```
Comment (vmarkovtsev) [inline server/athenian/api/models/web/team_tree.py:91]:
```suggestion
```
autogenerated

File: server/athenian/api/models/web/team_tree.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+    @property
+    def total_members(self) -> list[int]:
+        """Get the team members recursively included in the team tree."""
+        return self._total_members
+
+    @property
+    def members(self) -> list[int]:
+        """Get the directly contained members of the team."""
+        return self._members
```
Comment (vmarkovtsev) [inline server/athenian/api/models/web/team_tree.py:103]:
```suggestion
```
autogenerated

**Review by vmarkovtsev — CHANGES_REQUESTED** [review]

File: server/athenian/api/models/web/base_model_.py (hunk 2 above)
Comment (gaetano-guerriero) [reply server/athenian/api/models/web/base_model_.py:33]: done, anyway we could also evaulate to drop support for dikt['default_values'] and dikt['attribute_types'], annotations are enough

Comment (vmarkovtsev) [reply server/athenian/api/models/web/base_model_.py:33]: They are quite useful. In fact, dataclasses have their own version in `__dataclass_fields__`

**Review by vmarkovtsev — APPROVED** [review]

**Pull request merged**